### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by saadat-ali-awan**
### Project requirements
- Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
    - [UI example](https://www.youtube.com/watch?v=vkkgrhD6aXQ)
- Your console app should behave in the same way is in the example.
- Create a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain `Person`).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person id.
- In your `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method should do the following:
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### CODE REVIEW

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Saadat123456 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is failing rubocop linters, would you please fix them up and submit again?

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: App/app.rb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp.to_i
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    student = Student.new(age, 0, name, parent_permission: %w[Y y].include?(permission))
```
Comment (hiromataba) [inline App/app.rb:37]: - [ ] When creating a student, the data are not being displayed correctly they are misplaced, the name takes the age argument and the age is becoming 0 

image

File: App/app.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp.to_i
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
```
Comment (hiromataba) [inline App/app.rb:50]: - [ ] When creating a teacher, the data are not being displayed correctly they are misplaced, the name takes the age argument and the age is becoming 0 

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @Saadat123456     :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — CHANGES_REQUESTED** [review]
- [ ] Please update the above changes :arrow_up:

File: App/app.rb (hunk 2 above)
Comment (DeliceLydia) [reply App/app.rb:50]: - [ ] This should be addressed

File: App/app.rb (hunk 1 above)
Comment (DeliceLydia) [reply App/app.rb:37]: - [ ] Please address this

File: App/app.rb [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    student = Student.new(age, 0, name, parent_permission: %w[Y y].include?(permission))
```
Comment (DeliceLydia) [inline App/app.rb:37]: - [ ] Please address this as suggested by the previous reviewer

File: App/app.rb [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
```
Comment (DeliceLydia) [inline App/app.rb:50]: - [ ] Please address this as suggested

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Saadat123456, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: App/app.rb (hunk 4 above)
Comment (saadat-ali-awan) [reply App/app.rb:50]: Can you kindly check it again? It is working on my side.
Screenshot from 2022-05-21 20-13-04

Comment (DeliceLydia) [reply App/app.rb:50]: - [ ] Please check what is wrong on my side I have the same issue or check if you pushed updated codes
t

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Saadat123456, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: App/app.rb (hunk 4 above)
Comment (saadat-ali-awan) [reply App/app.rb:50]: I have pushed all the changes and have counter-checked and run again on my side. Maybe the project you cloned is not the recent one. Can you kindly see into this? If you want I am available for the zoom call.

**Review by rukundoeric — APPROVED** [review]
Hi @Saadat123456,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 




Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Davidosky007 — COMMENTED** [review]
Hello @Saadat123456  👋,

_This is an additional review as a result of the second opinion request._

* Well done 👏 seems all issues have been resloved 👍 

As this was the only reason that your project was not approved it will be approved as soon as possible.

Keep rocking 🚀!

**Pull request merged**